### PR TITLE
Enable Scalajs For Syntax

### PR DESCRIPTION
**Pull request by Josef-Vonasek**
### Checklist
Please include the following checklist in your PR:

- [x] The documentation has been updated if necessary.
- [x] All code conforms to the [Scala](https://github.com/luna/enso/blob/master/doc/scala-style-guide.md), [Java](https://github.com/luna/enso/blob/master/doc/java-style-guide.md), [Rust](https://github.com/luna/enso/blob/master/doc/rust-style-guide.md) or [Haskell](https://github.com/luna/enso/blob/master/doc/haskell-style-guide.md) style guides as appropriate.
- [x] All code has been tested where possible.

### CODE REVIEW

**Comment by sylwiabr** [comment]
Please connect it to proper issue and fill PR description

File: common/scala/syntax/specialization/src/test/scala/org/enso/syntax/text/DocParserTests.scala [hunk 1]
```diff
@@ -17,7 +18,7 @@ class DocParserTests extends FlatSpec with Matchers {
     output match {
       case Result(_, Result.Success(value)) =>
         assert(value == result)
-        assert(value.show() == input)
```
Comment (mwu-tow) [inline common/scala/syntax/specialization/src/test/scala/org/enso/syntax/text/DocParserTests.scala]: Why this change? 
The original version seems to be more directly testing that parser/generator round-trips.  Why should it involve `Reader`?

Comment (Josef-Vonasek) [reply common/scala/syntax/specialization/src/test/scala/org/enso/syntax/text/DocParserTests.scala]: Because we need to replace `\r`, `\r\n` with `\n` and `\t` with ` `..

File: common/scala/syntax/definition/src/main/scala/org/enso/data/Pool.scala [hunk 2]
```diff
@@ -19,14 +19,14 @@ import scala.reflect.internal.util.WeakHashSet
...
-  private val astPool = WeakHashSet[T]()
+//  private val astPool = WeakHashSet[T]()
 
   /** Returns object from pool such that `object == t`, or puts t into pool
     * and returns it, if no such object is found.
     *
     * The asymptotic complexity is almost identical to the of HashSet.get (logN)
     * The function is thread safe and throws an error when t is null
     */
-  def get(t: T): T = synchronized(astPool.findEntryOrUpdate(t))
```
[the remark this answers is not in the document]
Comment (Josef-Vonasek) [reply common/scala/syntax/definition/src/main/scala/org/enso/data/Pool.scala]: For now I have removed the pool altogether. Although 30% memory improvement is nice, proper solution that would work for scalajs would likely involve wrapping ECMA6 WeakSet.

File: build.sbt [hunk 3]
```diff
@@ -153,40 +158,63 @@ val jmh = Seq(
 //// Sub-Projects ////
 //////////////////////
 
-lazy val logger = (project in file("common/scala/logger"))
```
[the remark this answers is not in the document]
Comment (Josef-Vonasek) [reply build.sbt]: Such helper would need to be a macro sadly. Which I am not very keen on writing.
